clang/lib/CIR/Dialect/Transforms/FlattenCFG.cpp

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements pass that inlines CIR operations regions into the parent
+// function region.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "clang/CIR/Dialect/IR/CIRDialect.h"
+#include "clang/CIR/Dialect/Passes.h"
+#include "clang/CIR/MissingFeatures.h"
+
+using namespace mlir;
+using namespace cir;
+
+namespace {
+
+struct CIRFlattenCFGPass : public CIRFlattenCFGBase<CIRFlattenCFGPass> {
+
+  CIRFlattenCFGPass() = default;
+  void runOnOperation() override;
```

Comment:
Curious about the order of this function/why it isn't inline like the ones in CIRScopeOpFlattening.

Reply:
Yeah, it was just jarring that the type belwo it was all 'inline', but this one was out of line and below the other, with no real references between the two.

Reply:
> so I've been wondering if I should make a pass at canonicalizing/organizing the order of things in the incubator.

I think that could be a worthwhile investment.

I, too, have been trying to keep things in the same order as in the incubator, and have also found the lack of any reasonableness in the order to be frustrating.